packages/components/badge/src/badge.vue

```diff
@@ -39,6 +39,11 @@ const content = computed<string>(() => {
   return `${props.value}`
 })
 
+const isShowBadge = computed(() => {
+  const isShowVal = +content.value === 0 ? props.showZero : content.value
+  return !props.hidden && (isShowVal || props.isDot)
+})
```

Comment:
```suggestion
```
You can't tell the `content.value` is a string or number type, please move the judgment to the computed `content`.

```
const content = computed<string>(() => {
  if (props.isDot) return ''

  if (props.value === 0 && !props.showZero) return ''

  if (isNumber(props.value) && isNumber(props.max)) {
    return props.max < props.value ? `${props.max}+` : `${props.value}`
  }
  return `${props.value}`
})
```